### PR TITLE
A series of patches for Android

### PR DESCRIPTION
**Pull request by cwhuang**
These patches fix building issues of Android on the master branch.
Must be built with https://github.com/01org/libva/pull/37

Tested OK on Android-x86 7.1 cm branch.

### CODE REVIEW

**Comment by xuguangxin** [comment]
@LindaYu17 please help review this

File: src/Android.mk [hunk 1]
```diff
@@ -24,60 +24,16 @@
...
-	intel_memman.c		\
-	object_heap.c		\
-	intel_media_common.c		\
-	$(NULL)
+LOCAL_SRC_FILES := $(source_c)
 
 LOCAL_MODULE := i965_drv_video
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := SHARED_LIBRARIES
+LOCAL_MODULE_RELATIVE_PATH := dri
```
Comment (LindaYu17) [inline src/Android.mk]: could you please put libva*.so and i965_drv_video.so to /vendor/lib folder, all vendor provided driver/hal should come to this folder.

**Comment by cwhuang** [comment]
Move the libs path to /vendor as request.

**Comment by xhaihao** [comment]
@LindaYu17 Please help to review the updated patches and approve the changes if you are fine with the patches.

**Comment by LindaYu17** [comment]
+1

**Pull request merged**